core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -256,19 +251,41 @@ class ZkUtils(val zkClient: ZkClient,
   }
 
   /**
-   * Register brokers with v2 json format (which includes multiple endpoints).
+   * Register brokers with v3 json format (which includes multiple endpoints and rack).
```

Comment:
OK, I think this has been discussed before and I apologise for asking again. Why do we need to write v2 at all? With the exception of 0.9.0.0, the parsing code reads any version and ignore fields that it doesn't know about, right? Is it to make it possible to upgrade from 0.9.0.0 (even if that will break old clients anyway)?